backend/auth/handlers.go

```diff
@@ -0,0 +1,340 @@
+		return
+	}
+	c.JSON(http.StatusOK, gin.H{})
+}
+
+func ChangeUserAccountPasssword(c *gin.Context) {
+	logger.WebUILog.Infoln("change user password")
+	username := c.Param("username")
+	var user configmodels.User
+	err := c.ShouldBindJSON(&user)
+	if err != nil {
+		logger.AuthLog.Errorln(err.Error())
+		c.JSON(http.StatusBadRequest, gin.H{"error": errorInvalidDataProvided})
+		return
+	}
+	if user.Password == "" {
+		c.JSON(http.StatusBadRequest, gin.H{"error": errorMissingPassword})
+		return
+	}
+	if !validatePassword(user.Password) {
+		c.JSON(http.StatusBadRequest, gin.H{"error": errorInvalidPassword})
+		return
+	}
+	dbUser, err := fetchUser(username)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": errorRetrieveUserAccount})
+		return
+	}
+	if dbUser == nil {
+		c.JSON(http.StatusNotFound, gin.H{"error": errorUsernameNotFound})
+		return
+	}
+	hashedPassword, err := bcrypt.GenerateFromPassword([]byte(user.Password), bcrypt.DefaultCost)
+	if err != nil {
+		logger.AuthLog.Errorln(err.Error())
+		c.JSON(http.StatusInternalServerError, gin.H{"error": errorUpdateUserAccount})
+		return
+	}
+	dbUser.Password = string(hashedPassword)
+	filter := bson.M{"username": dbUser.Username}
+	_, err = dbadapter.WebuiDBClient.RestfulAPIPost(UserAccountDataColl, filter, configmodels.ToBsonM(dbUser))
+	if err != nil {
+		logger.DbLog.Errorln(err.Error())
+		c.JSON(http.StatusInternalServerError, gin.H{"error": errorUpdateUserAccount})
+		return
+	}
+	c.JSON(http.StatusOK, gin.H{})
+}
+
+func Login(jwtSecret []byte) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		var inputUser configmodels.User
+		err := c.ShouldBindJSON(&inputUser)
+		if err != nil {
+			logger.AuthLog.Errorln(err.Error())
+			c.JSON(http.StatusBadRequest, gin.H{"error": errorInvalidDataProvided})
+			return
+		}
+		if inputUser.Username == "" {
+			c.JSON(http.StatusBadRequest, gin.H{"error": errorMissingUsername})
+			return
+		}
+		if inputUser.Password == "" {
+			c.JSON(http.StatusBadRequest, gin.H{"error": errorMissingPassword})
+			return
+		}
+		dbUser, err := fetchUser(inputUser.Username)
+		if err != nil {
+			c.JSON(http.StatusInternalServerError, gin.H{"error": errorRetrieveUserAccount})
+			return
+		}
+		if dbUser == nil {
+			c.JSON(http.StatusUnauthorized, gin.H{"error": errorIncorrectCredentials})
+			return
+		}
+		if err = bcrypt.CompareHashAndPassword([]byte(dbUser.Password), []byte(inputUser.Password)); err != nil {
+			c.JSON(http.StatusUnauthorized, gin.H{"error": errorIncorrectCredentials})
+			return
+		}
+		jwt, err := generateJWT(dbUser.Username, dbUser.Permissions, jwtSecret)
+		if err != nil {
+			logger.AuthLog.Errorln(err.Error())
+			c.JSON(http.StatusInternalServerError, gin.H{"error": errorLogin})
+			return
+		}
+		c.JSON(http.StatusOK, gin.H{"token": jwt})
+	}
+}
+
+func IsFirstAccountIssued() (bool, error) {
+	users, err := fetchUsers()
+	if err != nil {
+		return false, err
+	}
+	return len(users) > 0, nil
+}
```

Comment:
Instead of fetching all users, we should have a db query to get the number of users 

```go
numUsers, err := numUsers()
```